### PR TITLE
Remaining Yaml additions for 2022 Odrive CAN

### PR DESCRIPTION
**Pull request by jessieramseyer**

### CODE REVIEW

File: scripts/generate_can.py [hunk 1]
```diff
@@ -103,82 +103,83 @@ def camel_to_snake_case(value):
...
-                        is_signed=signal['is_signed'],
-                        scale=signal['scale'],
-                        offset=signal['offset'],
-                        minimum=signal['min'],
-                        maximum=signal['max'],
-                        unit=signal['unit'],
-                        choices=signal['values'] if 'values' in signal else None,
-                        comment=signal['comment'],
-                        receivers=message['receivers']
+            if 'signals' in message:
```
Comment (wmmc88) [inline scripts/generate_can.py:106]: @jessieramseyer plz dont introduce whitespace and formatting changes. makes diff harder to review

@cindyli-13 should add a quick ci stage to normalize formatting for python

File: can/uwrt_mars_rover_can_2022.yaml [hunk 2]
```diff
@@ -315,8 +326,8 @@ uwrt_mars_rover_can_bus:
             3: CONTROL_MODE_POSITION_CONTROL
           scale: 1.0
           offset: 0.0
-          min: 0.0
-          max: 1.0
+          min: -2147483647.0
```
Comment (wmmc88) [inline can/uwrt_mars_rover_can_2022.yaml:329]: shouldnt this be -2147483648?

File: can/uwrt_mars_rover_can_2022.yaml [hunk 3]
```diff
@@ -334,293 +345,670 @@ uwrt_mars_rover_can_bus:
             8: INPUT_MODE_Tuning
           scale: 1.0
           offset: 0.0
-          min: 0.0
-          max: 1.0
-  - ODRIVE_getVbusVoltage_axis0:
-      comment: get vbus voltage
-      id: 23
-      size: 4
+          min: -2147483647.0
```
Comment (wmmc88) [inline can/uwrt_mars_rover_can_2022.yaml:348]: ditto

File: can/uwrt_mars_rover_can_2022.yaml [hunk 4]
```diff
@@ -41,7 +41,7 @@ uwrt_mars_rover_can_bus:
           scale: 1.0
```
Comment (wmmc88) [inline can/uwrt_mars_rover_can_2022.yaml:41]: @cindyli-13 is this file automatically normalized?

**Review by wmmc88 — CHANGES_REQUESTED** [review]
missing the generated dbcs

File: can/uwrt_mars_rover_can_2022.yaml (hunk 2 above)
Comment (jessieramseyer) [reply can/uwrt_mars_rover_can_2022.yaml:329]: I'm not sure. I just ran the generation script and it populated this file with those values...

File: scripts/generate_can.py (hunk 1 above)
Comment (jessieramseyer) [reply scripts/generate_can.py:106]: Yeah I agree, the diff is really hard to look at. How would I fix it? I'm not sure which formatting changes I made

Comment (wmmc88) [reply scripts/generate_can.py:106]: its whitespace changes. the number of spaces in the file per indent level has changed

File: can/uwrt_mars_rover_can_2022.yaml (hunk 4 above)
Comment (cindyli-13) [reply can/uwrt_mars_rover_can_2022.yaml:41]: currently no

File: can/uwrt_mars_rover_can_2022.yaml (hunk 2 above)
Comment (cindyli-13) [reply can/uwrt_mars_rover_can_2022.yaml:329]: iirc the script leaves the max magnitude value as the SNA (signal not available) value, so it shows `min` as one smaller

**Review by cindyli-13 — APPROVED** [review]
Thanks for working on this :)

LGTM, but looks like the clang format job is failing (see [this PR](https://github.com/uwrobotics/MarsRoverFirmware/pull/419))? The `generate_can.py` script should auto format the generated files but could you double check if it actually did (if it didn't can you manually format each of the .h / .c / .cpp files)? After that you can merge this PR and the corresponding FW repo PR.

I know you were still having issues with SavvyCAN on Windows 11, but a potential and important task you could do (or someone else) is to write a Nucleo test app to try out sending and receiving CAN messages between a Nucleo and an ODrive, using our CAN framework. The setup would be `ODrive <- CAN -> Nucleo <- USB -> your computer`, and you can use printfs from the nucleo to a terminal screen on your computer to check if the CAN messages are correct. This task would be a separate PR.

cc @jahnavithota2011

**Pull request merged**